### PR TITLE
check for tests leaving views behind

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6045/

File: js/client/modules/@arangodb/test-utils.js [hunk 1]
```diff
@@ -274,21 +281,30 @@ function performTests (options, testList, testname, runFn, serverOptions, startS
             return (name[0] !== '_'); // exclude system collections from the comparison
           });
 
-          if (delta.length !== 0) {
+          let deltaViews = diffArray(viewsBefore, viewsAfter).filter(function(name) {
+            return ! ((name[0] === '_') || (name === "compact") || (name === "election")
+                     || (name === "log")); // exclude system/agency collections from the comparison
```
Comment (joerg84) [inline js/client/modules/@arangodb/test-utils.js]: s/collection/view

File: js/client/modules/@arangodb/test-utils.js [hunk 2]
```diff
@@ -248,22 +252,25 @@ function performTests (options, testList, testname, runFn, serverOptions, startS
             break;
           }
         }
-
         if (pu.arangod.check.instanceAlive(instanceInfo, options) &&
             healthCheck(options, serverOptions, instanceInfo, customInstanceInfos, startStopHandlers)) {
           continueTesting = true; 
 
           // Check whether some collections were left behind, and if mark test as failed.
```
Comment (joerg84) [inline js/client/modules/@arangodb/test-utils.js]: collections or views

File: js/client/modules/@arangodb/test-utils.js [hunk 3]
```diff
@@ -274,21 +281,30 @@ function performTests (options, testList, testname, runFn, serverOptions, startS
             return (name[0] !== '_'); // exclude system collections from the comparison
           });
 
-          if (delta.length !== 0) {
+          let deltaViews = diffArray(viewsBefore, viewsAfter).filter(function(name) {
+            return ! ((name[0] === '_') || (name === "compact") || (name === "election")
+                     || (name === "log")); 
+            return (name[0] !== '_');
```
Comment (mchacki) [inline js/client/modules/@arangodb/test-utils.js]:
```suggestion
```
this line is unreachable and covered by the one above

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6111/

**Pull request merged**